Project 3/Task 1/block_quantize.py

```diff
@@ -0,0 +1,126 @@
+import sys
+import math
+import numpy as np
+import cv2
+import matplotlib.pyplot as plt
+import re
+import Utility as util
+from os import listdir,path
+
+'''
+Quantizes a block using the number of bins
+Returns a formatted string for output to a file which details the information in the format:
+< FrameNumber, (X coordinate, Y coordinate) , Y Channel Value, Number Of Occurances >
+'''
+def quantizeBlock(block, bins, frameNum, x, y):
+    minValue = np.amin(block)
+    maxValue = np.amax(block)
+    samples, valuesPerBin = np.linspace(minValue, maxValue, bins, endpoint=False, retstep=True)
+    samples += 0.5*valuesPerBin
+    samples = map((lambda v: round(v, 7)), samples)
+    occurancesInFrame = dict.fromkeys(samples, 0) #dictionary to contain how many times a value occurs
+
+    for val in np.nditer(block):
+        if valuesPerBin == 0:
+            bottomBin = minValue
+        else:
+            bottomBin = minValue + (math.floor((val - minValue) / valuesPerBin) + 0.5) * valuesPerBin
+
+        if bottomBin >= maxValue:
+            bottomBin = minValue + (bins - 0.5)*valuesPerBin
+
+        bottomBin = round(bottomBin, 7)
+        occurancesInFrame[bottomBin] += 1
```

Comment:
The dictionary you are creating, is is mapping keys from our samples to 0, or setting the values to zero? I had an issue where given a range that fromkeys() function doesnt generate the key at the top of the range, which is why the old code had +1.